### PR TITLE
Content negotiation improvements

### PR DESCRIPTION
**Pull request by james-emerton**
These changes fix a couple of content negotiation issues.

The original code expected media type parameters to be separated with commas, when they should actually be separated with semicolons.

I've also added support for the quality (q) parameter to allow the client to better influence the rendered that is selected.

### CODE REVIEW

**Comment by james-emerton** [comment]
This appears to be failing because of some flake8 warnings on a file I didn't touch.

**Comment by tomchristie** [comment]
I'm not currently supporting flask-api, at least until the work on the [Django REST framework 3 kickstarter](https://www.kickstarter.com/projects/tomchristie/django-rest-framework-3) is complete, so I'm afraid I'm going to have to pass on reviewing this for the moment. Sorry to not be more help. One inline note added, tho.

File: flask_api/tests/test_mediatypes.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 class MediaTypeParsingTests(unittest.TestCase):
     def test_media_type_with_params(self):
-        media = MediaType('application/xml; schema=foobar, q=0.5')
+        media = MediaType('application/xml; schema=foobar; q=0.5')
```
Comment (tomchristie) [inline flask_api/tests/test_mediatypes.py]: Probably correct, mind linking me to an RFC or similar to refresh myself on this?

Comment (james-emerton) [inline flask_api/tests/test_mediatypes.py]: I found that [RFC 2616 Section 14](http://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html) seemed to indicate that parameters are separated with commas, but the grammar is not very clear.
I also referred to [some other W3C documentation](http://www.w3.org/Protocols/HTTP/HTRQ_Headers.html) for confirmation.

Comment (tomchristie) [inline flask_api/tests/test_mediatypes.py]: Thanks @james-emerton appreciate that.
Apologies for not being active on Flask API at the moment - I'd love to get back to it, but a case of focusing on one thing at a time right now.

**Comment by james-emerton** [comment]
I've moved the relevant commits to their own branch and opened #36

**Pull request closed without merging**